packages/@aws-cdk/aws-codedeploy/lib/server/deployment-group.ts

```diff
@@ -306,12 +306,12 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupBase {
     });
 
     const resourceIdentifiers = new cdk.ResourceIdentifiers(this, {
-      arn: arnForDeploymentGroup(this.application.applicationName, resource.refAsString),
-      name: resource.refAsString,
+      arn: arnForDeploymentGroup(this.application.applicationName, resource.ref),
+      name: resource.ref,
       arnComponents: {
         service: 'codedeploy',
         resource: 'deploymentgroup',
-        resourceName: `${this.application.physicalName.value}/${this.physicalName.value}`,
+        resourceName: `${this.application.physicalName}/${this.physicalName}`,
```

Comment:
Like we talked about - this should probably be `this.application.applicationName` instead of `physicalName` (ditto in other places).